src/Compilers/VisualBasic/Portable/Symbols/SynthesizedSymbols/SynthesizedParameterSymbol.vb

```diff
@@ -233,6 +233,15 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Symbols
                 Return _name
             End Get
         End Property
+
+        Friend NotOverridable Overrides Sub AddSynthesizedAttributes(compilationState As ModuleCompilationState, ByRef attributes As ArrayBuilder(Of SynthesizedAttributeData))
+            Dim compilation = Me.DeclaringCompilation
+            If Type.ContainsTupleNames() AndAlso
+                compilation.HasTupleNamesAttributes AndAlso
```

Comment:
Yes, I think they can be removed. The logic in `BindTupleType` (`LookupTupleType` in VB) should already have reported errors if either the attribute or the string types are missing.
I'd leave a comment though.
Also, let's double-check with @VSadov, who made those early checks un-conditional (https://github.com/dotnet/roslyn/commit/75a042bf).